core/services/wifi/WifiManager.py

```diff
@@ -209,3 +205,51 @@ async def disconnect(self) -> None:
             await self.wpa.send_command_disconnect()
         except Exception as error:
             raise ConnectionError("Failed to disconnect from wifi network.") from error
+
+    async def enable_saved_networks(self) -> None:
+        """Enable saved networks."""
+        try:
+            saved_networks = await self.get_saved_wifi_network()
+            for network in saved_networks:
+                await self.wpa.send_command_enable_network(network.networkid)
+            await self.wpa.send_command_save_config()
+            await self.wpa.send_command_reconfigure()
+        except Exception as error:
+            raise RuntimeError("Failed to enable saved networks.") from error
+
+    async def auto_reconnect(self, seconds_before_reconnecting: float) -> None:
+        """Re-enable all saved networks if disconnected for more than specified time.
+        When a connection is made, using the 'select' wpa command, all other saved networks are disabled, to ensure
+        connection to the desired network. This prevents the system from reconnecting to one of those networks when
+        away from the last connected one.
+
+        This watchdog re-enable all connections after a specified time to ensure the system don't stay disconnected
+        forever even if there are known networks nearby.
+
+        Arguments:
+            seconds_before_reconnecting {float} -- Seconds to wait disconnected before enabling all saved networks.
+        """
+        seconds_disconnected = 0.0
+        networks_reenabled = False
+        was_connected = True
+        while True:
+            status = await self.status()
+            is_connected = "wpa_state" in status and not status["wpa_state"] in ["DISCONNECTED", "SCANNING"]
+            if was_connected and not is_connected:
+                time_disconnection = time.time()
+                was_connected = False
+                logger.debug("Lost connection.")
+            if not was_connected and not is_connected:
+                seconds_disconnected = time.time() - time_disconnection
+                logger.debug(f"Seconds since disconnection: {seconds_disconnected}.")
+            if not networks_reenabled and seconds_disconnected >= seconds_before_reconnecting:
+                await self.enable_saved_networks()
+                await self.wpa.send_command_reconnect()
+                networks_reenabled = True
+                logger.debug("Enabled all networks.")
+            if not was_connected and is_connected:
+                seconds_disconnected = 0
+                networks_reenabled = False
+                was_connected = True
```

Comment:
this two booleans appears to be working as enum. It may worth to think a bit more and think if that is the case to improve the logic and readability of the code.

Reply:
A enum would definitely help readability here. Will do.